applications/vila_live/Dockerfile

```
@@ -18,57 +18,83 @@
+# NOTE: on Arm64 this will take between 1-1.5 hours
+# NOTE: MAX_JOBS is set to 4 to avoid running out of memory
+RUN MAX_JOBS=4 python3 -m pip install --no-cache-dir --no-build-isolation flash-attn==2.7.*
 
-# Copy over app-specific requirements
-COPY applications/vila_live/requirements.txt applications/vila_live/llm-awq.patch /tmp/
+# ----------------------------------------------------------------------------
+# Clone and install AWQ based on the GPU's compute capacity
+# ----------------------------------------------------------------------------
+# Copy over patch file to patch llm-awq so that it works with transformers 4.48
+COPY applications/vila_live/llm-awq.patch /tmp/
 
-# Clone AWQ and VILA repositories and install them based on the GPU's compute capacity
+# Clone and install AWQ based on the GPU's compute capacity
 WORKDIR /workspace
 ARG COMPUTE_CAPACITY
-RUN git clone https://github.com/mit-han-lab/llm-awq \
-    && cd llm-awq \
-    && git checkout 41d7e704932b7eaf8b3dcfb21e1fe49fd296d894 \
+RUN git clone https://github.com/mit-han-lab/llm-awq && \
+    cd llm-awq && \
+    git checkout 41d7e704932b7eaf8b3dcfb21e1fe49fd296d894 && \
     # patching this version of llm-awq so that it works with transformers 4.48
-    && git apply /tmp/llm-awq.patch \
+    git apply /tmp/llm-awq.patch && \
     # Remove torch and transformers from pyproject.toml to avoid conflicts with installed versions
-    && sed -i '/torch/d' pyproject.toml \
-    && sed -i '/transformers/d' pyproject.toml \
-    && python3 -m pip install --no-cache-dir -e . \
-    && cd awq/kernels \
-    && TORCH_CUDA_ARCH_LIST="${COMPUTE_CAPACITY}+PTX" \
-    MAX_JOBS=$(nproc) python setup.py install \
-    && cd ../.. \
-    && git clone https://github.com/Efficient-Large-Model/VILA.git \
-    && cd VILA \
-    && git checkout ec7fb2c264920bf004fd9fa37f1ec36ea0942db5 \
+    sed -i '/torch/d' pyproject.toml && \
+    sed -i '/transformers/d' pyproject.toml && \
+    python3 -m pip install --no-cache-dir -e . && \
+    cd awq/kernels && \
+    TORCH_CUDA_ARCH_LIST="${COMPUTE_CAPACITY}+PTX" MAX_JOBS=$(nproc) python setup.py install && \
+    cd ../../..
+
+# ----------------------------------------------------------------------------
+# Clone and install VILA repository
+# ----------------------------------------------------------------------------
+RUN git clone https://github.com/Efficient-Large-Model/VILA.git && \
+    cd VILA && \
+    git checkout ec7fb2c264920bf004fd9fa37f1ec36ea0942db5 && \
     # Remove torch and transformers from pyproject.toml to avoid conflicts with installed versions
-    && sed -i '/torch/d' pyproject.toml \
-    && sed -i '/transformers/d' pyproject.toml \
-    && python3 -m pip install --no-cache-dir --no-deps -e . \
-    && python3 -m pip install -r /tmp/requirements.txt
+    sed -i '/torch/d' pyproject.toml && \
+    sed -i '/transformers/d' pyproject.toml && \
+    python3 -m pip install --no-cache-dir --no-deps -e . && \
+    cd ..
```

Each RUN command runs in a new shell starting from last [WORKDIR](https://docs.docker.com/reference/dockerfile/#workdir), so trailing `cd` is a no-op:
```suggestion
    python3 -m pip install --no-cache-dir --no-deps -e .
```